src/python/pants/reporting/zipkin_reporter.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import requests
+from py_zipkin import Encoding
+from py_zipkin.transport import BaseTransportHandler
+from py_zipkin.zipkin import zipkin_span
+
+from pants.reporting.reporter import Reporter
+
+
+class HTTPTransportHandler(BaseTransportHandler):
+  def __init__(self, endpoint, encoding):
+    self.endpoint = endpoint
+    self.encoding = encoding
+
+  def get_max_payload_bytes(self):
+    return None
+
+  def send(self, payload):
+    try:
+      requests.post(
+        # Twitter endpoint 'https://zipkin.twitter.biz/api/v1/spans',
+        # Local endpoint 'http://localhost:9411/api/v1/spans',
+        self.endpoint,
+        data=payload,
+        headers={'Content-Type': 'application/{}'.format(self.encoding)},
+      )
+    except Exception as err:
+      err
+
+
+class ZipkinReporter(Reporter):
+  """Reporter that implements Zipkin tracing .
+  """
+
+  def __init__(self, run_tracker, settings, endpoint, encoding):
+    super(ZipkinReporter, self).__init__(run_tracker, settings)
+    # We keep track of connection between workunits and spans
+    self._workunits_to_spans = {}
+    self.encoding = encoding
+    # Create a transport handler
+    self.handler = HTTPTransportHandler(endpoint, encoding)
+
+  def open(self):
+    pass
+
+  def close(self):
+    pass
+
+  def start_workunit(self, workunit):
+    """Implementation of Reporter callback."""
+    # Check if it is the first workunit
+    first_span = not self._workunits_to_spans
+    if first_span:
+      span = zipkin_span(
+        service_name='pants_v1',
```

Comment:
Maybe just `pants`? Or you could look at the `WorkUnit`'s labels, and use something like `pants goal` or `pants task`?

(Same below)

Reply:
`WorkUnit` can have several labels, and not only `goal` or `task`. Why we may want to use only these two?

Reply:
Goal and task are useful container WorkUnits, we often want to see how long a goal took, or how long a task took. Everything else is a bit ad-hoc and varies from task to task - we may want to label some of them specially, but for now some logic like:

if has goal label:
  name = "pants goal"
elif has task label:
  name = "pants task"
else:
  name = "pants workunit"

would probably be useful :)